client/src/components/Panels/ToolPanel.vue

```diff
@@ -35,21 +33,16 @@ const emit = defineEmits<{
 }>();
 
 const arePanelsFetched = ref(false);
-const defaultPanelView = ref("");
 const toolStore = useToolStore();
-const { currentPanelView, isPanelPopulated } = storeToRefs(toolStore);
+const { currentPanelView, defaultPanelView, isPanelPopulated, panelViews } = storeToRefs(toolStore);
 
 const query = ref("");
-const panelViews = ref<Record<string, PanelView> | null>(null);
 const showAdvanced = ref(false);
 
 onMounted(async () => {
-    await axios
-        .get(`${getAppRoot()}api/tool_panels`)
-        .then(async ({ data }) => {
-            const { default_panel_view, views } = data;
-            defaultPanelView.value = default_panel_view;
-            panelViews.value = views;
+    await toolStore
```

Comment:
A regular try/catch/finally will read better when using await. I know it was like this before but this seems like a good opportunity to improve it :)

Reply:
Thanks @davelopez, yes, I'll make a point of using try/catch/finally instead of just awaits from now on